bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/ExperienceFragmentImpl.java

```diff
@@ -0,0 +1,183 @@
+        resolver = request.getResourceResolver();
+        pageManager = currentPage.getPageManager();
+    }
+
+    @Override
+    public String getFragmentPath() {
+        return fragmentPath;
+    }
+
+    @Override
+    public String getLocalizedFragmentPath() {
+        String localizedFragmentPath = null;
+        Page xfPage = pageManager.getPage(fragmentPath);
+        if (xfPage != null) {
+            Page xfLocalizedPage = getXfLocalizedPage(xfPage, currentPage);
+            if (xfLocalizedPage != null) {
+                localizedFragmentPath = xfLocalizedPage.getContentResource().getPath();
+            } else {
+                localizedFragmentPath = xfPage.getContentResource().getPath();
+            }
+        } else {
+            LOGGER.debug("Experience Fragment variation not found at path:{}", fragmentPath);
+        }
+        return localizedFragmentPath;
+    }
+
+    @NotNull
+    @Override public String getExportedType() {
+        return request.getResource().getResourceType();
+    }
+
+    /**
+     * Returns the localized page of the experience fragment page, based on the content page and the localization properties.
+     *
+     * @param xfPage the experience fragment page
+     * @param sitePage the content page
+     * @return
+     */
+    private Page getXfLocalizedPage(@NotNull Page xfPage, @NotNull Page sitePage) {
+        Resource siteRoot = resolver.getResource(localizationRoot);
+        if (siteRoot == null) {
+            return null;
+        }
+        String xfLocalizationRoot = StringUtils.replace(localizationRoot, CONTENT_ROOT, EXPERIENCE_FRAGMENTS_ROOT, 1);
+        Resource xfRoot = resolver.getResource(xfLocalizationRoot);
+        if (xfRoot == null) {
+            return null;
+        }
+
+        String xfPageLocalizationString = getLocalizationString(xfPage, xfRoot, localizationDepth); // e.g. us/en
+        String sitePageLocalizationString = getLocalizationString(sitePage, siteRoot, localizationDepth); // e.g. us/es
+        if (StringUtils.isEmpty(xfPageLocalizationString)) {
+            return null;
+        }
+        if (StringUtils.isEmpty(sitePageLocalizationString)) {
+            return null;
+        }
+
+        String xfLocalizationSrcPath = StringUtils.joinWith(PATH_DELIMITER, xfLocalizationRoot, xfPageLocalizationString); // e.g. /content/experience-fragments/my-site/us/en
+        String xfLocalizationDestPath = StringUtils.joinWith(PATH_DELIMITER, xfLocalizationRoot, sitePageLocalizationString); // e.g. /content/experience-fragments/my-site/us/es
+
+        String xfLocalizedPagePath = StringUtils.replace(xfPage.getPath(), xfLocalizationSrcPath, xfLocalizationDestPath, 1);
+        return pageManager.getPage(xfLocalizedPagePath);
+    }
+
+    /**
+     * Returns the localization string of the given page based on the localization root and depth.
+     *
```

Comment:
In fact the method is ignorant of localization.
It simply makes path replacement.

I would advice to reflect this in its name.